### PR TITLE
sudo sessions timeout

### PR DESCRIPTION
**Pull request by dominikl**
See Ticket [12972](http://trac.openmicroscopy.org/ome/ticket/12972)
Previously if a session for another user was created a zero timeout was set, with this PR now the same timeout as the existing session is used.
Test: Code review, check that the import as other user (will initiate a sudo session) still works.

### CODE REVIEW

File: components/blitz/src/omero/gateway/Connector.java [hunk 1]
```diff
@@ -855,7 +855,9 @@ public Connector call() throws Exception {
                 p.name = userName;
                 p.eventType = "Sessions";
                 ISessionPrx prx = entryEncrypted.getSessionService();
-                Session session = prx.createSessionWithTimeout(p, 0L);
+                Session session = prx.getSession(secureClient.getSessionId());
+                long timeout = session.getTimeToIdle().getValue();
+                session = prx.createSessionWithTimeout(p, timeout);
```
Comment (sbesson) [inline components/blitz/src/omero/gateway/Connector.java]: This method apparently sets the `timeToLive` and you are setting the `timeToIdle` from the first session, you might want to use [createSessionWithTimeouts(principal, long, long)](http://downloads.openmicroscopy.org/omero/5.1.3/api/ome/services/sessions/SessionBean.html#createSessionWithTimeouts(ome.system.Principal,%20long,%20long)) instead.

**Comment by dominikl** [comment]
Thanks @sbesson , didn't notice that; fixed now.

**Comment by sbesson** [comment]
Using the OMERO.insight from https://ci.openmicroscopy.org/view/DEV/job/OMERO-DEV-merge-build/66/ (OS X), imported an image as another user

```
$ bin/omero sessions -q who
 name    | group           | logged in               | agent          | timeout | hitCount | lastAccessTime      
---------+-----------------+-------------------------+----------------+---------+----------+---------------------
...
 user-11 | read-annotate-1 | 2015-08-21 07:35:56     | createSession  | 10 min. | 28       | 2015-08-21 07:35:57 
 root    | system          | 2015-08-21 07:34:03     | OMERO.insight  | 10 min. | 63       | 2015-08-21 07:36:04 
07:41:43 
 user-11 | read-annotate-1 | 2015-08-21 07:34:52     | createSession  | 10 min. | 337      | 2015-08-21 07:35:56 
(13 rows)
```

And the sudoed sessions timeout after 10 minutes of inactivity as expected. :+1:

**Comment by dominikl** [comment]
Just agent "createSession" looks a bit weird, wonder where that comes from...

**Comment by sbesson** [comment]
This agent is set by https://github.com/openmicroscopy/openmicroscopy/blob/v5.1.3/components/server/src/ome/services/sessions/SessionBean.java#L119. We might want to add extra methods allowing to control the agent that could be used by clients like CLI, Insight etc

**Comment by dominikl** [comment]
Manually rebased, see #4086

--no-rebase

**Comment by joshmoore** [comment]
Did any of the points (or the need to manually lookup the timeout) get listed as an RFE anywhere?

**Comment by jburel** [comment]
you mean a card? ticket is http://trac.openmicroscopy.org/ome/ticket/12972

**Comment by joshmoore** [comment]
Sorry, no. I meant the other RFEs.

**Comment by sbesson** [comment]
See https://trello.com/c/azP0udku

**Comment by dominikl** [comment]
--rebased-to #4086

**Pull request merged**